src/studio/src/designer/backend/Controllers/DatamodelsController.cs

```diff
@@ -173,6 +172,30 @@ public async Task<IActionResult> AddXsd(string org, string repository, [FromForm
             return Created(fileName, jsonSchema);
         }
 
+                /// <summary>
+        /// Upload an XSD (for datamodelling repos).
+        /// </summary>
+        /// <remarks>
+        /// This operation will use the new datamodelling library to convert the XSD into a JSON schema
+        /// </remarks>
+        /// <param name="org">The short name of the application owner.</param>
+        /// <param name="repository">The name of the repository to which the file is being added.</param>
+        /// <param name="filePath">The path to the XSD to use</param>
+        [Authorize]
+        [HttpGet("addFromRepo")]
+        public async Task<IActionResult> AddXsdFromRepo(string org, string repository, string filePath)
+        {
```

Comment:
It'd be nice to add a integration test for new endpoint in DesignerTests project in Controllers/DataModelsController